src/main/java/tc/oc/pgm/rotation/MapPoll.java

```diff
@@ -222,7 +218,16 @@ private int countVotes(Set<UUID> uuids) {
    * @return The picked map to play after the vote
    */
   MapInfo finishVote() {
-    MapInfo picked = getMostVotedMap();
+    // Cache vote totals, so we don't have to count for each compare
+    for (Entry<MapInfo, Set<UUID>> map : votes.entrySet()) {
+      finalTotals.put(map.getKey(), countVotes(map.getValue()));
```

Comment:
Instead of making a new map for the cache, change the Set<UUID> to a specific collection that keeps track of double-votes and caches the vote count.